tools/background_removal/background_removal.xml

```diff
@@ -0,0 +1,55 @@
+<tool id="background_removal" name="Background Removal" version="0.1">
+    <description>with scikit-image</description>
+    <requirements>
+        <requirement type="package" version="0.24.0">scikit-image</requirement>
+    </requirements>
+    <command detect_errors="aggressive">
+        <![CDATA[
+        python "$__tool_directory__/background_removal.py" $input_image $filter $radius $output
+        ]]>
+    </command>
+    <inputs>
+        <param name="input_image" type="data" format="tiff, jpg, jpeg, png, tif" label="Input Image"/>
+        <param name="filter" type="select" label="Select Filter">
+            <option value="rolling_ball">Rolling-Ball Algorithm</option>
+            <option value="dog">Difference of Gaussians</option>
+            <option value="top_hat">Top-Hat Filter</option>
+        </param>
+        <param name="radius" type="float" label="Radius" value="20"/>
+    </inputs>
+    <outputs>
+        <data name="output" format="tiff" label="Background Substraction Output"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_image" value="img.png"/>
+            <param name="filter" value="rolling_ball"/>
+            <param name="radius" value="20"/>
+            <output name="output" value="out_rolling_ball.tiff" ftype="tiff"/>
```

Comment:
Can you please use a suitable compare method for the tests, like `image_diff`?
https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output

Alternatively, I'd suggest using one of the macros we have pre-defined for images. See this for example:
https://github.com/BMCV/galaxy-image-analysis/blob/5a1ff1a3a5df15f75c95734af3ead420de536c0b/tools/2d_simple_filter/filter.xml#L116

Reply:
I just added the macro tested locally and everything seems to work. Let me know if this is done correctly, hope I understood it right